scripts/Crawlers/OSFCrawler.py

```diff
@@ -114,6 +114,27 @@ def _download_files(self, link, current_dir, inner_path, d, annex, sizes):
                     file_size       = int(annex_info_dict['size'])
                 sizes.append(file_size)
 
+    def _download_components(self, components_list, current_dir, d, annex, dataset_size):
+        # Loop through each available components and download their files
+        for component in components_list:
+            component_title = self._clean_dataset_title(component['attributes']['title'])
+            component_inner_path = os.path.join('components', component_title)
+            os.makedirs(os.path.join(current_dir, component_inner_path))
+            self._download_files(
+                component['relationships']['files']['links']['related']['href'],
+                os.path.join(current_dir, component_inner_path),
+                component_inner_path,
+                d,
+                annex,
+                dataset_size
+            )
```

Comment:
what happens if the component has a sub-component? not sure if it's even possible, but if it is we should support it or at least print a warning that it is not supported.

Reply:
good point. Did not think of that possibility. I'll check if this is possible with OSF but I don't see why this would not be possible. Thanks!

Reply:
confirmed. We can create subcomponents. Will work on modifying the code this PM to accommodate for that. :)